server.js

```diff
@@ -1,33 +1,82 @@
 import express from 'express'
 import bodyParser from 'body-parser'
 import cors from 'cors'
+import netflixData from './data/netflix-titles.json'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+
+console.log(netflixData)
+
+
+
+
+// Defines the port the app will run on. 
 const port = process.env.PORT || 8080
 const app = express()
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
+
+
 app.get('/', (req, res) => {
   res.send('Hello world')
 })
 
-// Start the server
+//this get the whole api
+app.get('/api', (req, res) => {
+  res.json(netflixData)
+})
+
+//filters with query and endpoint search
+app.get('/shows', (req, res) => {
+  const searchString = req.query.search;
+  // console.log(searchString)
+
+  let filteredShows = netflixData;
+  // console.log(filteredShows)
+
+
+  if (searchString) {
+    // if the (searchString) exist:
+
+    //filters the data for a title, country, description & director
+    filteredShows = filteredShows.filter(item => {
+      const itemTitle = item.title.toString();
+      const itemCountry = item.country.toString();
+      const itemDescrpition = item.description.toString();
+      const itemDirector = item.director.toString();
+
+      return itemTitle.toLowerCase().includes(searchString) ||
+        itemCountry.toLowerCase().includes(searchString) ||
+        itemDescrpition.toLowerCase().includes(searchString) ||
+        itemDirector.toLowerCase().includes(searchString);
+
+    })
+    res.json(filteredShows)
+  }
+})
+
+//filters with id
+app.get('/shows/id/:id', (req, res) => {
+  const id = req.params.id
+  // console.log(id)
+  const showId = netflixData.filter((item) => item.show_id === +id)
+  res.json(showId)
+})
+
+//filters with release year
+app.get('/shows/year/:year', (req, res) => {
```

Comment:
I think it would be nice to merge this filter into the main `/shows` endpoint further up. So you could do `/shows?year=2019` and do the filtering in there.